### PR TITLE
set_security_groups_from_manifest: don't scrape cf output, use cf curl

### PR DESCRIPTION
**Pull request by risicle**
What
----

It looks like cf's output has changed enough to make the regex not work correctly anymore, so use json from `cf curl` instead.

We could probably take some further steps to ensure this has gone through ok, e.g. try to read back the configuration after an attempted update, but we want to get this out.

How to review
-------------

Look at `dev01`.

---

🚨⚠️ Please do not merge this pull request via the GitHub UI ⚠️🚨

### CODE REVIEW

**Review by cadmiumcat — APPROVED** [review]
I can see it works 👍 
But another pair of 👀 would be great

File: concourse/scripts/set_security_groups_from_manifest.rb [hunk 1]
```diff
@@ -47,17 +47,10 @@ def existing_security_groups
...
-    security_groups = `cf security-groups`
-    abort security_groups unless $CHILD_STATUS.success?
-
-    security_groups.each_line do |line|
-      if line =~ /\A#\d+\s+(\S+)\s*/
-        groups << Regexp.last_match(1)
-      end
-    end
-    groups
+    JSON.parse(response)["resources"].map {|sg| sg["name"]}
```
Comment (AP-Hunt) [inline concourse/scripts/set_security_groups_from_manifest.rb:53]: I like this. It's simple, concise, and self-explanatory

**Review by AP-Hunt — APPROVED** [review]

**Comment by risicle** [comment]
Incidentally this is why I don't like low-level heavily mocked unit tests. They don't catch things that are broken IRL and they make it painful to fix.

**Comment by risicle** [comment]
omg robocop don't even get me started.

File: concourse/scripts/set_security_groups_from_manifest.rb (hunk 1 above)
Comment (risicle) [reply concourse/scripts/set_security_groups_from_manifest.rb]: Too concise for rubocop it turns out

**Pull request merged**